kie-dmn-feel/src/main/java/org/kie/dmn/feel/util/EvalHelper.java

```diff
@@ -285,4 +291,106 @@ public static String lcFirst(final String name) {
         return name.toLowerCase().charAt( 0 ) + name.substring( 1 );
     }
 
+    /**
+     * Compares left and right operands using the given predicate and returns TRUE/FALSE accordingly
+     *
+     * @param left
+     * @param right
+     * @param ctx
+     * @param op
+     * @return
+     */
+    public static Boolean compare(Object left, Object right, EvaluationContext ctx, BiPredicate<Comparable, Comparable> op) {
+        if ( left == null || right == null ) {
+            return null;
+        } else if ( (left instanceof Period && right instanceof Period ) ) {
+            // periods have special compare semantics in FEEL as it ignores "days". Only months and years are compared
+            Period lp = (Period) left;
+            Period rp = (Period) right;
+            Integer l = lp.getYears() * 12 + lp.getMonths();
+            Integer r = rp.getYears() * 12 + rp.getMonths();
```

Comment:
Doh. Stupid me.. your reply is exactly what I was thinking it should be doing; but for some reason thought it wasn't.